sklearn/gaussian_process/kernels.py

```diff
@@ -33,6 +33,21 @@
 from sklearn.externals.funcsigs import signature
 
 
+def _check_length_scale(X, length_scale):
+    if np.iterable(length_scale):
```

Comment:
Not sure, but arithmetic operations involving a 0d-array and a Python scalar produce a Python scalar; those involving a 0d-array and a >0d numpy array produce a numpy array... I doubt there'll be any issue except if there's any type checking (I doubt it), and the tests should tell us if anything obvious breaks.